### PR TITLE
Remove DSA signatures

### PR DESCRIPTION
**Pull request by beurdouche**
For when the WG decides it is ok...

### CODE REVIEW

**Comment by davegarrett** [comment]
I only see a couple spots left that need updating:
1) Update the version number to FIPS 186-4 in "Negotiated Groups" to match the updated DSS reference.
2) Update the version number to FIPS 180-4 in "Digital Signing" to match the current SHS reference, while you're at it.

**Comment by davegarrett** [comment]
To cross-ref: this PR fixes issue #222 and obsoletes issue #59

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -2204,7 +2186,9 @@ signature
   The values indicate anonymous signatures, RSASSA-PKCS1-v1_5
   {{RFC3447}}, DSA {{DSS}}, and ECDSA {{ECDSA}}, respectively.  The
   "anonymous" value is meaningless in this context but used in
-  {{server-key-share}}.  It MUST NOT appear in this extension.
+  {{server-key-share}}. It MUST NOT appear in this extension.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: minor tweak suggestion: merge these two sentences to drop the "it" for slightly better clarity, e.g.
> The
>  "anonymous" value is meaningless in this context, but used in
>  {{server-key-share}}, and MUST NOT appear in this extension.

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -2204,7 +2186,9 @@ signature
...
+  The use of DSA is deprecated. The dsa_RESERVED value MUST NOT 
+  be offered or negotiated by any implementation.
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: Actually, disregard my previous note on this part of the text. Don't improve the anon bit; remove it. It appears to be obsolete, and this value isn't used anymore. The Server Key Message in that section is fundamentally different now. Change the "anonymous" value in the enum to "anon_RESERVED" and change this text to something like this, instead:
> The use of DSA and the anonymous value are deprecated.
> The anon_RESERVED and dsa_RESERVED values MUST
> NOT be offered or negotiated by any implementation.


>   anonymous_RESERVED placeholder value is meaningless in this context but used in
>  {{server-key-share}}.  It MUST NOT appear in this extension.
+  {{server-key-share}}. It MUST NOT appear in this extension.
+  The use of DSA is deprecated. The dsa_RESERVED value MUST NOT 
+  be offered or negotiated by any implementation.

Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: Actually, disregard my previous note on this part of the text. Don't improve the anon bit; remove it. It appears to be obsolete, and this value isn't used anymore. The Server Key Message in that section is fundamentally different now. Change the "anonymous" value in the enum to "anon_RESERVED" and change this text to something like this, instead:
> The use of DSA and the anonymous value are deprecated.
> The anon_RESERVED and dsa_RESERVED values MUST
> NOT be offered or negotiated by any implementation.

**Comment by ekr** [comment]
@seanturner  do we have consensus to merge this PR?

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -2202,9 +2184,10 @@ hash
 signature
 : This field indicates the signature algorithm that may be used.
   The values indicate anonymous signatures, RSASSA-PKCS1-v1_5
-  {{RFC3447}}, DSA {{DSS}}, and ECDSA {{ECDSA}}, respectively.  The
-  "anonymous" value is meaningless in this context but used in
-  {{server-key-share}}.  It MUST NOT appear in this extension.
+  {{RFC3447}}, DSA {{DSS}}, and ECDSA {{ECDSA}}, respectively.
+  The use of DSA and anonymous are deprecated. The anonymous_RESERVED
+  and dsa_RESERVED values MUST NOT be offered or negotiated by any
+  implementation.
```
Comment (ekr) [inline draft-ietf-tls-tls13.md]: A little birdie tells me that we're not going to actually forbid DSA in the spec right now, so please remove this last sentence. Also "is deprecated" I believe.

File: draft-ietf-tls-tls13.md [hunk 4]
```diff
@@ -3084,6 +3051,7 @@ The input to the client and server Finished messages may not be
 the same because the server's Finished does not include the client's
 Certificate and CertificateVerify message.
 
+
```
Comment (ekr) [inline draft-ietf-tls-tls13.md]: This seems superfluous.

File: draft-ietf-tls-tls13.md (hunk 3 above)
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: The "are deprecated" is correct. It's "X and Y are" OR "X or Y is".

If we're going to continue to wait for more complaining about DSA on the list, just change the "MUST NOT" to a "SHOULD NOT" for now. This does mean two parts, though, as the anonymous value should stay a "MUST NOT". So, let's just make this bit:

> The anonymous_RESERVED value was used internally in prior versions of TLS and MUST NOT be offered or negotiated by any implementation of any TLS version. The dsa_RESERVED value is deprecated as of TLS 1.3 and SHOULD NOT be offered or negotiated by any implementation.

File: draft-ietf-tls-tls13.md [hunk 5]
```diff
@@ -739,28 +732,12 @@ implementations MUST accept both without parameters and with NULL parameters.
...
-refer to the standard, and it uses "DSS" in the code point definitions for
-historical continuity.
-
 All ECDSA computations MUST be performed according to ANSI X9.62 {{X962}}
 or its successors.  Data to be signed/verified is hashed, and the
 result run directly through the ECDSA algorithm with no additional
 hashing.  The default hash function is SHA-1 {{SHS}}.  However, an
 alternative hash function, such as one of the new SHA hash functions
-specified in FIPS 180-2 may be used instead if the certificate
+specified in FIPS 180-4 {{SHS}} may be used instead if the certificate
```
Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: Just noticed this. This "may" should probably be changed to a "SHOULD".

File: draft-ietf-tls-tls13.md (hunk 3 above)
Comment (ekr) [inline draft-ietf-tls-tls13.md]: I agree that "X or Y are deprecated" but this says "The use of X and Y are deprecated" so the subject is "use" not "X and Y".

Comment (davegarrett) [inline draft-ietf-tls-tls13.md]: On Wednesday, September 16, 2015 06:25:55 pm ekr wrote:
> I agree that "X or Y are deprecated" but this says "The use of X and Y are deprecated" so the subject is "use" not "X and Y".

That makes sense, though both sound right to me... Now I don't know which is correct, but I think it's the current version. English is annoying. :p

Regardless, the anon is a MUST NOT and dsa isn't (yet), so it's getting changed either way.

**Comment by beurdouche** [comment]
Updated, rebased, merged conflicts with PSS patch and force pushed.

**Comment by ekr** [comment]
Merged

**Pull request merged**